devito/function.py

```diff
@@ -913,7 +941,7 @@ def avg(self, p=None, dims=None):
 
     # Pickling support
     _pickle_kwargs = TensorFunction._pickle_kwargs +\
-        ['space_order', 'shape', 'dimensions']
+        ['space_order', 'shape_global', 'dimensions']
```

Comment:
Only pickle local shape?

Reply:
the global shape should be pickled -- the local shape is inferred individually by each MPI rank from the global shape and from the decomposition. This change fixes exactly this (pickle global, not local)